product_abc_classification/models/product_product.py

```diff
@@ -19,3 +19,33 @@ class ProductProduct(models.Model):
         column2="profile_id",
         index=True,
     )
+    abc_classification_profile_updatable_from_category = fields.Boolean(default=True)
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        res = super().create(vals_list)
+        for rec in res:
+            if (
+                not rec.abc_classification_profile_ids
+                and rec.categ_id.abc_classification_profile_ids
+            ):
+                rec.abc_classification_profile_ids = (
+                    rec.categ_id.abc_classification_profile_ids
+                )
+        return res
+
+    def write(self, vals):
+        if not vals.get("abc_classification_profile_ids") and vals.get("categ_id"):
+            category = self.env["product.category"].browse(vals["categ_id"])
+            vals["abc_classification_profile_ids"] = [
+                (6, 0, category.abc_classification_profile_ids.ids)
+            ]
+        return super().write(vals)
+
+    @api.onchange("categ_id")
```

Comment:
```suggestion
    @api.onchange("categ_id", "abc_classification_profile_ids")
```

Reply:
if the user changes the profiles, I want to keep what the user updates